crates/services/gas_price_service/src/v1/uninitialized_task.rs

```diff
@@ -198,10 +198,15 @@ where
             self.block_stream,
         );
 
-        let starting_recorded_height = self
-            .gas_price_db
-            .get_recorded_height()?
-            .unwrap_or(BlockHeight::from(latest_block_height));
+        let starting_recorded_height = match self.gas_price_db.get_recorded_height()? {
+            Some(height) => height,
+            None => {
+                let mut storage_tx = self.gas_price_db.begin_transaction()?;
```

Comment:
Where do you commit this transaction? Also, maybe it makes sense to allow customization of the gas price height from CLI? For example: you want to sync with the network from genesis block, but you only want to calculate the gas price starting the latest block height, to avoid huge debt because data was not posted during previous blocks.

Reply:
> Where do you commit this transaction?

Hmm. I'll double check that it makes sense. I got the test to pass so I assumed it was okay but you're right. Might be able to move this to the sync section.

>Also, maybe it makes sense to allow customization of the gas price height from CLI? 

Yeah. I think that's a good idea.